### ISSUE TITLE
In POSIX sh, rmdir flags besides -p are undefined

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1455]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
rmdir --parent /a/b/c
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
"In POSIX sh, rmdir flags besides -p are undefined"

The only acceptable flag for `rmdir` under POSIX sh is `-p` ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/rmdir.html)), `--parent` is a GNU synonym for `-p` not defined by POSIX. There are other flags defined by GNU's `rmdir` (such as `--ignore-fail-on-non-empty`) which are also not POSIX.

**Comment by contivero** [comment]
After opening this and uploading a fix to my fork, I realized it might not be the appropriate way (you can check it [here](https://github.com/contivero/shellcheck/commit/cc0e16cff912de6e2e567cf04fa916a45ca26d99)). I approched it as the other similar fixes I did, but this one is not technically a bashism (GNUism?), so I'm not sure if I should leave it inside ShellSupport.hs, or if I should move it to the Commands.hs file.

**Comment by koalaman** [comment]
Like you say, bad flags to `export`, `ulimit` and other builtins are a bashism or shell support issue, while this is a more general userland issue. ShellCheck is currently only aware of which shell you're targeting, it doesn't know which userland. 

The shell and userland are not necessarily connected:
* Debian scripts target POSIX sh (or dash), but can rely on full access to GNU utilities.
* MacOS scripts run on GNU Bash, but can not rely on any GNU utilities

There's not currently any support for trying to check whether e.g. a GNU/Linux script will run on macOS. This would be nice and has been briefly discussed a few times before (e.g. #674), but it's not currently supported and there's quite a few details to be worked out.

**Comment by contivero** [comment]
I'll focus on builtins then. You may close this, or leave it open in case it is eventually supported, however you see fit. Thanks for the detailed explanation! 🙂
